### PR TITLE
fixes #955 adding client recipe to cookbook

### LINKED ISSUE #955: A Cookbook recipe of using a Finagle client
This has been asked several times and we should have a cookbook example already.

### PR DESCRIPTION
**Pull request by erikwj**

### CODE REVIEW

File: docs/src/main/tut/cookbook.md [hunk 1]
```diff
@@ -351,9 +351,33 @@ val policy: Cors.Policy = Cors.Policy(
...
+import com.twitter.finagle.http._
+import com.twitter.util.{Await, Future}
+
+val host = "api.nasa.gov"
+val client: Service[Request, Response] = Http.client.withTls(host).newService(s"$host:443")
+val request = Request(Method.Get, "/planetary/apod?api_key=NNKOjkoul8n1CH18TWA9gwngW1s1SmjESPjNoUFo")
+val response: Future[Response] = client(request)
+
+Await.result(response.onSuccess { rep: Response =>
+  //println("GET success: " + rep.contentString)
```
Comment (rpless) [inline docs/src/main/tut/cookbook.md]: We might want to remove this commented out line.

**Comment by rpless** [comment]
@erikwj Thanks for putting this together! I left a couple comments. Curious to hear your thoughts.

File: docs/src/main/tut/cookbook.md [hunk 2]
```diff
@@ -351,9 +351,33 @@ val policy: Cors.Policy = Cors.Policy(
...
+documentation to get a full understanding of the configuration options for the client like its [loadbalancing][loadbalancing] abilities. A simple example is given below.
+
+```tut:silent
+import com.twitter.finagle.Http
+import com.twitter.finagle.http._
+import com.twitter.util.{Await, Future}
+
+val host = "api.nasa.gov"
+val client: Service[Request, Response] = Http.client.withTls(host).newService(s"$host:443")
+val request = Request(Method.Get, "/planetary/apod?api_key=NNKOjkoul8n1CH18TWA9gwngW1s1SmjESPjNoUFo")
```
[the remark this answers is not in the document]
Comment (vkostyukov) [reply docs/src/main/tut/cookbook.md]: Can we use `DEMO_KEY` instead? This https://api.nasa.gov/planetary/apod?api_key=DEMO_KEY works for me. They have some limits about the `DEMO_KEY` usage but I think we should be good: https://api.nasa.gov/api.html#demo_key-rate-limits

**Comment by vkostyukov** [comment]
Thanks, @erikwj! This LGTM (after @rpless's concern about the API key is addressed).

File: docs/src/main/tut/cookbook.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (rpless) [reply docs/src/main/tut/cookbook.md]: :+1: to using the demo key.

File: docs/src/main/tut/cookbook.md [hunk 3]
```diff
@@ -351,9 +351,33 @@ val policy: Cors.Policy = Cors.Policy(
 val corsService: Service[Request, Response] = new Cors.HttpFilter(policy).andThen(service)
 ```
 
+### Creating a Finagle Client
```
Comment (vkostyukov) [inline docs/src/main/tut/cookbook.md]: Can you also add a TOC item for this? I don't think we generate automatically.

File: docs/src/main/tut/cookbook.md (hunk 1 above)
Comment (erikwj) [reply docs/src/main/tut/cookbook.md]: @rpless Agree on all above will update

File: docs/src/main/tut/cookbook.md (hunk 3 above)
Comment (erikwj) [reply docs/src/main/tut/cookbook.md]: will do

**Pull request merged**